test/complex.jl

```diff
@@ -1,3 +1,72 @@
 using Zygote, Test
 
-@test gradient(x -> abs(x)*exp(im*angle(x)), 10+20im)[1] ≈ 1
+# Gradients are defined as the backpropagation of the number 1. If C is seen as R^2, this means the first column of the transpose of the jacobian.
+# For C -> R functions f(x+iy), the Jacobian is (df/dx;df/dy), and so the gradient is df/dx + i df/dy
+# For complex valued functions, the gradient is the gradient of the real part. For C -> C functions u+iv=f(x+iy), the gradient is du/dx + i du/dy
+# For holomorphic functions, ∇f (as defined above) = du/dx + i du/dy = du/dx - i dv/dx = conj(f'), with f' the complex derivative
+
+fs_C_to_R = (real,
+             imag,
+             abs,
+             abs2,
+             z -> abs(z)*cos(im*angle(z)),
+             # z->z.re,
+             # z->z.im,
+             # z->2z.re + 3z.im,
```

Comment:
You could just use the `real` and `imag` functions here

Reply:
Sure, the point was to test these as well

Reply:
If we add them they will trivially be the same as `real`/`imag`, so it's probably not too important to test them independently.

Reply:
That's not clear to me why it needs a specific implementation (it should just be a struct access?), but I can remove them, sure

Reply:
The default rule for struct access creates an "adjoint struct" for `Complex` like `(re = grad, im = nothing)`; we have to define it explicitly so that it produces a new complex number instead (and also so that it handles complex `grad` by dropping the imaginary part).